segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StorageEventProcessor.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.server.containers;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.util.BufferView;
+import io.pravega.segmentstore.server.ContainerEventProcessor;
+import io.pravega.segmentstore.storage.chunklayer.AbstractTaskQueueManager;
+import io.pravega.segmentstore.storage.chunklayer.GarbageCollector;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Implementation of {@link AbstractTaskQueueManager} that uses {@link io.pravega.segmentstore.server.ContainerEventProcessor.EventProcessor}
+ * as underlying implementation.
+ * This class acts as adaptor that converts calls on {@link io.pravega.segmentstore.server.ContainerEventProcessor.EventProcessor}
+ * into appropriate calls on {@link AbstractTaskQueueManager} and vice versa.
+ */
+@Slf4j
+public class StorageEventProcessor implements AbstractTaskQueueManager<GarbageCollector.TaskInfo> {
+
+    private final static GarbageCollector.TaskInfo.Serializer SERIALIZER = new GarbageCollector.TaskInfo.Serializer();
+
+    private final int containerID;
+    private final ContainerEventProcessor eventProcessor;
+    private final Function<List<GarbageCollector.TaskInfo>, CompletableFuture<Void>> callBack;
+    private final int maxItemsAtOnce;
+    private final String traceObjectId;
+    @Getter
+    private final ConcurrentHashMap<String, ContainerEventProcessor.EventProcessor> eventProcessorMap = new ConcurrentHashMap<>();
```

Comment:
`ContainerEventProcessorImpl` already has this map, for which this is creating a redundant map for the same. You can have here just a `Set<String>` with the names of the processors you need. Then, you could create a new method in the `ContainerEventProcessor` interface called `get(String processorName)` that returns the `CompletableFuture<EventProcessor>` in the case it exists or `null` otherwise. This will help this class and any other class with a similar use case to avoid keeping references to the `EventProcessor`s in maps like this.

Reply:
We need this lookup to translate StorageEventProcessor::addTask to processor.add on every call.
Looking up by calling get() will be too much overhead and will eat up time on core thread pool.
Also realistically there are very few entries in that map.